addons/payment_stripe/controllers/main.py

```diff
@@ -114,6 +114,64 @@ def stripe_webhook(self):
+                tx_sudo = request.env['payment.transaction'].sudo()._get_tx_from_feedback_data(
+                    'stripe', data
+                )
+                if self._verify_webhook_signature(tx_sudo.acquirer_id.stripe_webhook_secret):
+                    # Get all refunds for this charge
+                    refunds = charge['refunds']['data']
+                    has_more = charge['refunds']['has_more']
+                    while has_more:
+                        more_refunds = tx_sudo.acquirer_id._stripe_make_request(
+                            # The url provided by Stripe begin by /v1/, but it's already included in
+                            # _stripe_make_request()
+                            charge['refunds']['url'][4:],
+                            payload={'starting_after': refunds[-1].get('id')},
+                            method='GET',
+                        )
+                        refunds += more_refunds['data']
+                        has_more = more_refunds['has_more']
+
+                    # Process refunds we aren't aware of
+                    for refund in filter(lambda r: r['metadata'] == {}, refunds):
+                        # _get_tx_from_feedback_data() create the transaction if it doesn't exist
+                        refund_tx = request.env['payment.transaction'].sudo()._stripe_get_refund_tx(
+                            tx_sudo, refund
+                        )
+                        # Update the reference on Stripe
+                        refund = tx_sudo.acquirer_id._stripe_make_request(
+                            f'refunds/{refund.get("id")}',
+                            payload={'metadata[reference]': refund_tx.reference},
+                            method='POST'
+                        )
+                        _logger.info(
+                            "received refund response:\n%s", pprint.pformat(refund)
+                        )
+                        # We don't handle feedback data here because Stripe will send back
+                        # immediately a 'charge.refund.updated', with the reference of the
+                        # transaction that we just set, and it may result in a DB cursor error
+                        # trying to write on the same transaction at the same time.
+            elif event['type'] == 'charge.refund.updated': # handle update of all refunds
+                refund = event['data']['object']
+
+                refund.update(
+                    reference=refund['metadata']['reference']
+                )
+                # Check the source and integrity of the event
+                tx_sudo = request.env['payment.transaction'].sudo()._get_tx_from_feedback_data(
+                    'stripe', refund
+                )
+                if self._verify_webhook_signature(tx_sudo.acquirer_id.stripe_webhook_secret):
+                    request.env['payment.transaction'].with_user(
+                        SUPERUSER_ID
```

Comment:
for the chatter messages (& record creation) ?
should be more targeted in specific places of the payment module :eyes:

Reply:
I removed it from this PR, it should be ok when we'll merge this commit: https://github.com/odoo/odoo/pull/78137/commits/c6d740228dd114b6d57f629aa801d6f0cdb479e8 (cf. https://github.com/odoo/odoo/pull/78137)